dist/js/multiselect.js

```diff
@@ -1,755 +1,1812 @@
+ * @property {function} afterMoveUp - called after options are moved up inside a right side select
+ * @property {function} beforeMoveDown - called before options are moved down inside a right side select
+ * @property {function} afterMoveDown - called after options are moved down inside a right side select
+ * @property {function} fireSearch - called to determine if the filter function for a side should be activated
+ */
+
+/**
+ * The settings object containing all possible options and information to create a Multiselect.
+ * @typedef {CallbackFunctions|ElementNames|MultiselectOptions} SettingsObject
+ */
+
+/**
+ * Describes options where the user provides inputs to use as select filters
+ * @typedef {Object} SearchElements
+ * @property {string|jQuery} [left] - either html for or a jQuery element pointing to the input to use for filtering the left palette
+ * @property {string|jQuery} [right] - either html for or a jQuery element pointing to the input to use for filtering the right palette
+ */
+
+/**
+ * The relation between a input used as a filter
+ * and the select elements that are filtered by it
+ * @typedef {Object} FilterRelation
+ * @property {jQuery} $filterInput - the input that you enter text in so that the select is filtered
+ * @property {jQuery} $filteredSelects - the select elements filtered by the input (currently always all right elements)
+ */
+
+/**
+ * Element of the stack for the Undo/Redo functionality.
+ * @typedef {Object} StackElement
+ * @property {jQuery} $lastSource - the element where the options were before the move
+ * @property {jQuery} $lastDestination - the element where the options were after the move
+ * @property {jQuery} $movedOptions - which options were moved
+ */
+
+/**
+ * The stack for the Undo/Redo functionality
+ * @typedef {StackElement[]} Stack
+ */
+
+/**
+ * Representation for an option. (Used when replacing the multiselect items)
+ * @typedef {Object} OptionRep
+ * @property {string} name - the text of the option
+ * @property {string} value - the value of the option
+ */
+
+/**
+ * Representation for an optgroup. (Used when replacing the multiselect items)
+ * @typedef {Object} OptgroupRep
+ * @property {string} label - the text of the optgroup
+ * @property {OptionRep[]} contents - the contents of the optgroup, i.e. options
+ */
+
+/**
+ * Representation for the content of a whole select element (i.e. options and optgroups).
+ * @typedef {Object} SelectContent
+ * @property {OptionRep[]} options - the options without any optgroup in the select
+ * @property {OptgroupRep[]} optgroups - the optgroups in this select
+ */
+
+(
+    /**
+     * Registers multiselect with amd or directly with jQuery.
+     * @param {function} factory - the factory that creates the multiselect api
+     */
+    function (factory) {
+        /**
+         * Check if jQuery is accessible in this script, fail when not.
+         */
+        function checkForGlobaljQuery() {
```

Comment:
Functions for jQuery checking. Global jQuery check is only done when not using AMD.